tools-v2/pkg/cli/command/curvebs/delete/snapshot/snapshot.go

```diff
@@ -0,0 +1,158 @@
+ * Created Date: 2023-11-10
+ * Author: ZackSoul
+ */
+package snapshot
+
+import (
+	"time"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	snapshotutil "github.com/opencurve/curve/tools-v2/internal/utils/snapshot"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	listSnapshot "github.com/opencurve/curve/tools-v2/pkg/cli/command/curvebs/list/snapshot"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/spf13/cobra"
+)
+
+const (
+	snapshotExample = `$ curve bs delete snapshot`
+)
+
+type SnapShotCommand struct {
+	basecmd.FinalCurveCmd
+	snapshotAddrs []string
+	timeout       time.Duration
+
+	user   string
+	file   string
+	uuid   string
+	failed bool
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*SnapShotCommand)(nil)
+
+func NewSnapShotCommand() *cobra.Command {
+	return NewDeleteSnapShotCommand().Cmd
+}
+
+func NewDeleteSnapShotCommand() *SnapShotCommand {
+	snapShotCommand := &SnapShotCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "snapshot",
+			Short:   "delete snapshot in curvebs",
+			Example: snapshotExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&snapShotCommand.FinalCurveCmd, snapShotCommand)
+	return snapShotCommand
+}
+
+func (sCmd *SnapShotCommand) AddFlags() {
+	config.AddBsSnapshotCloneFlagOption(sCmd.Cmd)
+	config.AddHttpTimeoutFlag(sCmd.Cmd)
+	config.AddBsUserOptionFlag(sCmd.Cmd)
+	config.AddBsSnapshotIDOptionFlag(sCmd.Cmd)
+	config.AddBsPathOptionFlag(sCmd.Cmd)
+	config.AddBsSnapshotFailedDOptionFlag(sCmd.Cmd)
+}
+
+func (sCmd *SnapShotCommand) Init(cmd *cobra.Command, args []string) error {
+	snapshotAddrs, err := config.GetBsSnapshotAddrSlice(sCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS || len(snapshotAddrs) == 0 {
+		return err.ToError()
+	}
+	sCmd.snapshotAddrs = snapshotAddrs
+	sCmd.timeout = config.GetFlagDuration(sCmd.Cmd, config.HTTPTIMEOUT)
+	sCmd.user = config.GetBsFlagString(sCmd.Cmd, config.CURVEBS_USER)
+	sCmd.file = config.GetBsFlagString(sCmd.Cmd, config.CURVEBS_PATH)
+	sCmd.uuid = config.GetBsFlagString(sCmd.Cmd, config.CURVEBS_SNAPSHOT_ID)
+	sCmd.failed = config.GetBsFlagBool(sCmd.Cmd, config.CURVEBS_SNAPSHOT_FAILED)
+	header := []string{
+		cobrautil.ROW_SNAPSHOT_ID,
+		cobrautil.ROW_SNAPSHOT_NAME,
+		cobrautil.ROW_RESULT,
+	}
+	sCmd.SetHeader(header)
+	return nil
+}
+
+func (sCmd *SnapShotCommand) Print(cmd *cobra.Command, args []string) error {
+	return output.FinalCmdOutput(&sCmd.FinalCurveCmd, sCmd)
+}
+
+func (sCmd *SnapShotCommand) RunCommand(cmd *cobra.Command, args []string) error {
+	params := map[string]any{
+		snapshotutil.QueryAction: snapshotutil.ActionGetFileSnapshotList,
+		snapshotutil.QueryUser:   sCmd.user,
+		snapshotutil.QueryFile:   sCmd.file,
+		snapshotutil.QueryLimit:  snapshotutil.Limit,
+		snapshotutil.QueryOffset: snapshotutil.Offset,
+	}
+	if sCmd.failed {
+		params[snapshotutil.QueryStatus] = 5
```

Comment:
It might be better to change the hard code.

Reply:
What does it mean to change the hard code？

Reply:
For example, you can use static variable to record what 5 means.

Reply:
ok，understand，I will modify it.